Sources/CoalescedIndices.swift

```diff
@@ -0,0 +1,206 @@
+public struct CoalescedIndices<Elements: BidirectionalCollection>: Collection {
+	public struct Index: Comparable {
+		public static func ==(left: Index, right: Index) -> Bool {
+			return true
+		}
+
+		public static func <(left: Index, right: Index) -> Bool {
+			return true
+		}
+
+	}
+
+	public var elements: Elements
```

Comment:
Similar to `Collection.Indices`, unless we store an instance of every index (which is impractical), we would need the collection itself to advance the indices.

An alternative is imposing an `Index: SignedInteger` constraint. So we can advance the indices on our own. But in this case, we might as well use the `IndexSet` type.